src/app.js

```diff
@@ -1,89 +1,110 @@
-import React from "react"
-import { fetchWeather } from "./utils/api"
-import fetch from "whatwg-fetch"
-import _ from "lodash"
-import randomColor from "randomcolor";
 
-import WeekWeather from "./components/WeekWeather"
+import React, { useState, useEffect } from 'react'
+import { fetchWeather } from './utils/api'
+import randomColor from 'randomcolor'
+import WeekWeather from './components/WeekWeather'
 
-import "./css/style.styl";
+import './css/style.styl'
 
+const App = () => {
+  const [city, setCity] = useState('Bucharest')
+  const [country, setCountry] = useState('')
+  const [search, setSearch] = useState('Bucharest')
+  const [weather, setWeather] = useState([])
+  const [color, setColor] = useState('')
 
-export default class App extends React.Component{
-  constructor(props) {
-    super(props);
-    this.state = {
-      city: "Bucharest",
-      searchedCity: "Bucharest",
-      weekWeather: []
-    };
+  const [address, setAddress] = useState({})
+  const [coords, setCoords] = useState({})
+
+  const shortCoord = (coord) => {
+    return coord.substr(0, coord.indexOf('.') + 7)
   }
 
-  componentWillMount() {
-    this.getWeather();
+  const showPosition = (position) => {
+    const latlng = {}
+    latlng['latitude'] = shortCoord(position.coords.latitude.toString())
+    latlng['longitude'] = shortCoord(position.coords.longitude.toString())
+    // console.log(latlng)
+    setCoords(latlng)
+  }
+  
+  const getCoords = () => {
+    navigator.geolocation.getCurrentPosition(showPosition);
   }
 
-  getWeather(searchedCity = this.state.city) {
-    fetchWeather(searchedCity)
-      .then((response) => {
-        var weather = _.map(response.list, (dayWeather) => {
-          return {
-            dayWeather,
-            country: response.city.country,
-            city: response.city.name
-          }
-        });
-
-        this.setState({
-          weekWeather: weather,
-          city: this.state.searchedCity,
-          color: randomColor({luminosity: "dark", format: "hex"})
-        });
+  const getAddress = () => {
+    if(Object.keys(coords).length !== 0) {
+      const API_URL = `${process.env.REACT_APP_ADD_URL}q=${coords.latitude}+${coords.longitude}&key=${process.env.REACT_APP_ADD_KEY}`
+
+      fetch(API_URL)
+      .then(res => res.json())
+      .then(data => {
+        // console.log(data)
+        const addr = {}
+        if(data.results[0].components.city !== undefined)
```

Comment:
Condition can be simply `if(data.results[0].components.city)`
And also we need to make sure that data.result actually exists like `if(data.results && data.results[0].components.city)`